src/main/java/seedu/address/model/event/EventDate.java

```diff
@@ -47,6 +48,10 @@ public static boolean isValidDate(String test) {
         return true;
     }
 
+    public static boolean checkDateComponents(String date) {
+        return date.split("-").length == 3;
```

Comment:
This is good enough ah? Just a question

Reply:
This simply checks for the format, if it has 3 components year-month-day. Then the next check is BirthDate::isValidDate --> This checks for non-existent dates eg 2021-00-00, or 2021-02-30. Then BirthDate::isValidBirthDate checks is the dates is in the future.

Each case throws a different error message:
1. "Dates should be in YYYY-MM-DD format!"
2. "Date does not exist!"
3. "Date of birth cannot be in the future." (For birthdate only)

Reply:
Can refer to ParserUtil.java to see the changes I made the parseBirthDate and parseEventDate methods